### PR TITLE
chore: add installation tests

### PR DESCRIPTION
**Pull request by DominicKramer**
These tests are part of the system tests and verify that the correct type definitions
are packaged with the module to ensure the module can be used in Typescript projects.

### CODE REVIEW

**Comment by DominicKramer** [comment]
PTAL

File: .gitignore [hunk 1]
```diff
@@ -7,6 +7,7 @@
 .nyc_output
 build
 docs/
+google-cloud-logging-winston-*.tgz
```
Comment (kjin) [inline .gitignore]: Can this be just `*.tgz`?

File: src/types/global.d.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+
```
Comment (kjin) [inline src/types/global.d.ts]: Add Apache2 license

**Comment by DominicKramer** [comment]
@jinwoo These tests are needed when using hand written `.d.ts` files.

In particular, with those files present, `tsc` can compile the code and make associated `.d.ts` files.  However, it is possible that those files reference the manually constructed `.d.ts` files.  In that case, if the referenced manually written `.d.ts` files are not bundled with the code during `npm pack`, Typescript users of this library will get compilation errors because the Typescript compiler won't be able to find all of the `.d.ts` files that describe the types for this library.

For example, `index.d.ts` has type definitions that use the `Options` type that is defined in `types/core.d.ts`.  If `types/core.d.ts` is not bundled with this library, a Typescript user, during compilation, will get an error when the compiler reads `index.d.ts` because it doesn't have the definition for the `Options` type.

**Pull request merged**